### DISCUSSION TITLE
Fuente dolar Blue

### DISCUSSION THREAD

**Discussion by pnco200** [guidospadavecchia/DolarBot#1] [Preguntas] [answered]
Me encanta el bot esta muy bueno y es muy ingenioso, lo unico que note es que el dolar blue no se esta actualizando a tiempo, no se si la fuente no se actualiza o si el bot no la actualiza.

**Comment by guidospadavecchia** [comment, marked as answer]
Hola, ¿Cómo estás?

Ante todo gracias por usarlo.

Tenés razón respecto al tiempo que tarda en actualizar la información, y que tal vez no es del todo inmediata. Es algo que ya noté hace un tiempo. Por un lado, te cuento que DolarBot consume una API (provista muy generosamente por [Castrogiovanni20/api-dolar-argentina](https://github.com/Castrogiovanni20/api-dolar-argentina)), que a su vez consume información de DolarSí (https://www.dolarsi.com/). Es decir que, en el fondo, dependemos de que DolarSí actualice la data para que el bot muestre la información correcta. De hecho, podes corroborarlo ingresando al sitio de DolarSí, y vas a ver que la información del bot se condice con lo que muestra esa página.

Lamentablemente, no hay mucho que pueda hacer respecto a la frecuencia con la cual se actualizan los datos. Lo único que solucionaría esto es cambiar el sitio del cual sacar la información, para lo cual el mismo tendría que ofrecer una API pública a la cual yo le pueda pegar (como lo hace DolarSí), y requeriría una reingeniería importante del backend de DolarBot. De todos modos, no lo descarto.

Si conocés alguna otra API pública de donde sacar los valores, que tenga información de las distintas cotizaciones lo puedo implementar en una versión futura.

Abrazo
